src/EnergyPlus/InputProcessing/InputProcessor.cc

```diff
@@ -868,11 +868,7 @@ void InputProcessor::setObjectItemValue(EnergyPlusData &state,
             }
             if (is_AlphaBlank) AlphaBlank()(alpha_index) = true;
         } else if (field_type == "n") {
-            if (within_max_fields) {
-                findDefault(Numbers(numeric_index), schema_field_obj);
-            } else {
-                Numbers(numeric_index) = 0;
-            }
+            findDefault(Numbers(numeric_index), schema_field_obj);
```

Comment:
`findDefault` is a bool function. Should check here and set the value to 0 if it returns false.

Reply:
No need to check, Numbers(numeric_index) is set to 0 at top of function and if default does not exist for that field will return false and a 0 for that field.

    bool InputProcessor::findDefault(Real64 &default_value, json const &schema_field_obj)
    {
        auto const find_default = schema_field_obj.find("default");
        default_value = 0;
        if (find_default != schema_field_obj.end()) {
            // get default for autosize, int or real
        }
        return false;
    }

Reply:
I'm not always inclined to use the latest fancy C++, but I will just note that C++23 will have std::expected, and it has the potential to be deployed throughout our code some day in the future.  We're years from anything like that, so no reason to discuss it, but feel free to look it up if you are an interested student.